src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -1597,6 +1624,16 @@ def delete_flavor(self, flavor_id: UUID) -> None:
             except NoResultFound as error:
                 raise KeyError from error
 
+    def _purge_non_custom_flavors(self) -> None:
```

Comment:
maybe `internal` or `builtin` instead of `non_custom`?

Reply:
built_in are our non-integration flavors, internal feels like it doesn't capture the integrations, basically this nukes everything that is not called 'custom' - feels like the most fitting name, even if it is a bit ugly

Reply:
Good point, I was somehow confusing this with our actual builtin flavors. Not a big fan of `non_custom` but can't think of anything better either 👍